lib/matplotlib/ticker.py

```diff
@@ -2130,7 +2130,16 @@ def subs(self, subs):
                                  "found '%s'." % subs)
             self._subs = subs
         else:
-            self._subs = np.asarray(subs, dtype=float)
+            try:
+                self._subs = np.asarray(subs, dtype=float)
+            except ValueError as e:
+                raise ValueError("If subs is not None and not a string, "
+                                 "it must be a sequence of float.") from e
+            if self._subs.ndim != 1:
+                raise ValueError("If subs is not None and not a string, it "
```

Comment:
"A sequence passed to subs must be 1-dimensional, not {}-dimensional"

Reply:
I changed it.